### PR TITLE
MAINT-37615: Allow the space manager to enable or disable the chat of a space.

### PR DESCRIPTION
**Pull request by mdiha**
I added a property (isEnabled) to chat rooms and i added two methods setRoomEnabled and isRoomEnabled to update this property and retrieve its value.
I made sure that space's rooms enabled property is set at server start-up if not already initialized.

### CODE REVIEW

File: server-embedded/src/main/java/org/exoplatform/chat/server/ChatServer.java [hunk 1]
```diff
@@ -842,6 +842,33 @@ public ChatServer()
...
+
+    String room = ChatUtils.getRoomId(spaceId);
+    Boolean isEnabled = chatService.isRoomEnabled(room);
+
+    return Response.ok(isEnabled.toString());
+  }
+
+  @Resource
+  @Route("/updateRoomEnabled")
+  public Response.Content setRoomEnabled(String user, String token, String spaceId, Boolean enabled) {
```
Comment (ahamdi) [inline server-embedded/src/main/java/org/exoplatform/chat/server/ChatServer.java:861]: This action should restricted to space owner, space managers and spaces administrators IMO

File: server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/ChatMongoDataStorage.java [hunk 2]
```diff
@@ -534,6 +537,47 @@ public void setRoomName(String room, String name) {
...
+      try {
+        DBObject doc = cursor.next();
+        if (doc.get("isEnabled") != null) {
+          isEnabled = (StringUtils.equals(doc.get("isEnabled").toString(), "true"));
+        } else {
+          doc.put("isEnabled", true);
+          coll.save(doc, WriteConcern.UNACKNOWLEDGED);
+        }
+      } catch (MongoException me) {
+        LOG.warning(me.getCode() + " : " + room + " : " + me.getMessage());
```
Comment (ahamdi) [inline server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/ChatMongoDataStorage.java]: This should be an error and not a warning

**Review by ahamdi — CHANGES_REQUESTED** [review]

File: server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/ChatMongoDataStorage.java (hunk 2 above)
Comment (mdiha) [reply server-embedded/src/main/java/org/exoplatform/chat/services/mongodb/ChatMongoDataStorage.java]: Okay will fix it

File: server-embedded/src/main/java/org/exoplatform/chat/server/ChatServer.java (hunk 1 above)
Comment (mdiha) [reply server-embedded/src/main/java/org/exoplatform/chat/server/ChatServer.java:861]: Okay will fix it

File: application/src/main/webapp/vue-app/extension.js [hunk 3]
```diff
@@ -479,19 +491,84 @@ export function registerExternalExtensions(chatTitle) {
...
 
-  extensionRegistry.registerExtension('profile-extension', 'action', profileExtensionAction);
+  if (extensionRegistry) {
+    extensionRegistry.registerExtension('profile-extension', 'action', profileExtensionAction);
+  }
 
   document.dispatchEvent(new CustomEvent('profile-extension-updated', { detail: profileExtensionAction}));
 }
 
+export function registerExternalComponents(componentName) {
```
Comment (ahamdi) [inline application/src/main/webapp/vue-app/extension.js:501]:
```suggestion
export function registerExternalSpaceSettingComponents(componentName) {
```

**Review by ahamdi — APPROVED** [review]

**Pull request merged**